Core/include/Acts/Utilities/Zip.hpp

```diff
@@ -0,0 +1,50 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <tuple>
+
+namespace Acts {
+
+/// Function that allows to zip two ranges to be used in a range-based for loop.
+/// This is a very basic implementation for two ranges, but could relatively
+/// easily be extended with variadic templates.
+/// @tparam RA The first range type
+/// @tparam RB The second range type
+/// @param ra The first range
+/// @param rb The second range
+/// @note the behaviour is undefined if the ranges do not have equal range
+template <typename RA, typename RB>
+auto zip(RA &ra, RB &rb) {
+  using ItA = decltype(ra.begin());
+  using ItB = decltype(rb.begin());
+
+  struct It {
+    ItA a;
+    ItB b;
+
+    using reference =
+        std::tuple<typename ItA::reference, typename ItB::reference>;
+
+    // clang-format off
```

Comment:
Why?

Reply:
yeah, so this originiates from the `std::iterator_traits`, which defines some typedefs that iterators should have (https://en.cppreference.com/w/cpp/iterator/iterator_traits)
So the thing is, an iterator can represent a proxy-class (which we have with the `ComponentProxy`) with reference-semantics, than this type is not a real reference type, and we want to pass it by-value (otherwise we would probably get an "cannot bind reference to tempory"-like error)
Hope this makes sense^^

Probably one should also define the other typedefs required by the iterator-traits, but this is not necessary for a range-for loop...

Reply:
Sure, I got the typedefs, but what about the format off? 😅

Reply:
Ah sorry I missed that its related to the clang-format...
Yeah I somehow wanted these function to fit in one line each, but since there is no real reason for this, I changed it